frontend/test/__runner__/run_cypress_tests.js

```diff
@@ -73,15 +73,26 @@ const init = async () => {
   await generateSnapshots();
 
   console.log(chalk.bold("Starting Cypress"));
-  let commandLineConfig = `baseUrl=${server.host}`;
-  if (sourceFolder) {
-    commandLineConfig = `${commandLineConfig},integrationFolder=${sourceFolderLocation}`;
-  } else if (singleFile) {
-    commandLineConfig = `${commandLineConfig},testFiles=${singleFileName}`;
-  } else {
+  const baseConfig = { baseUrl: server.host };
+  const folderConfig = sourceFolder && {
+    integrationFolder: sourceFolderLocation,
+  };
+  const specsConfig = singleFile && { testFiles: singleFileName };
+  const ignoreConfig =
     // if we're not running specific tests, avoid including db and smoketests
-    commandLineConfig = `${commandLineConfig},ignoreTestFiles=**/metabase-{smoketest,db}/**`;
-  }
+    folderConfig || specsConfig
+      ? null
+      : { ignoreTestFiles: "**/metabase-{smoketest,db}/**" };
+
+  const config = {
+    ...baseConfig,
+    ...folderConfig,
+    ...specsConfig,
+    ...ignoreConfig,
+  };
```

Comment:
`JSON.stringify` removes `undefined` properties (not `null` though), so you can do something like this to make things a little more compact/readable:

```
const config = {
  baseUrl: server.host,
  integrationFolder: sourceFolder ? sourceFolderLocation : undefined,
  testFiles: singleFile ? singleFileName : undefined,
  ignoreTestFiles: (sourceFolder || singleFile) ? undefined : '**/metabase-{smoketest,db}/**'
};
```

Reply:
Neat!
I didn't know that it completely removes `undefined` but doesn't remove `null`. Thanks.

While I agree that this is definitely more succinct, I'm afraid it can become harder to read (especially if we start adding more key/value pairs and more conditional logic).